etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateUniversalServiceIdentifier.java

```diff
@@ -0,0 +1,85 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.custom;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation.CustomFhirTransformation;
+import gov.hhs.cdc.trustedintermediary.external.hapi.HapiHelper;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.ServiceRequest;
+import org.hl7.fhir.r4.model.StringType;
+
+/**
+ * Override the Name of Coding System (OBR-4.3) and Alternate Identifier (OBR-4.4) In the Universal
+ * Service Identifier (OBR-4)
+ */
+public class UpdateUniversalServiceIdentifier implements CustomFhirTransformation {
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args) {
+        Bundle bundle = (Bundle) resource.getUnderlyingResource();
+        var serviceRequests = HapiHelper.resourcesInBundle(bundle, ServiceRequest.class);
+
+        serviceRequests.forEach(
+                it -> {
+                    var allCodings = it.getCode().getCoding();
+                    var codingSystemContainer =
+                            getCodingSystemContainer(allCodings, args.get("checkValue"));
+
+                    if (codingSystemContainer == null) {
+                        // we're only interested in coding that matches the checkValue argument
+                        return;
+                    }
+
+                    // check for the coding system label and create or override it
+                    var codingSystemLabel =
+                            codingSystemContainer.getExtensionByUrl(
+                                    HapiHelper.EXTENSION_CODING_SYSTEM);
+                    if (codingSystemLabel == null) {
+                        codingSystemLabel = new Extension();
```

Comment:
I'd extract this code and the altCodingExtension code into their own methods (Including the null checks) so that they can be easily reused in the future

Reply:
Done! It's on the same class for now.